sys/arm64/include/cheric.h

```diff
@@ -44,6 +46,13 @@ trapframe_set_elr(struct trapframe *tf, uintcap_t elr)
 	if (cheri_getsealed(elr))
 		elr = cheri_unseal(elr, sentry_unsealcap);
 
+#ifdef __ARM_MORELLO_PURECAP_BENCHMARK_ABI
+	/* Assume that kernel ELR addresses are C64 */
+	if (elr > VM_MAXUSER_ADDRESS) {
```

Comment:
The current implementation is that trapframe_set_elr should not be used for benchmark ABI processes, and that should be extended to the kernel.

Reply:
Ok, I can directly update `tf->elr` in trap.c conditionally, I did not notice that we avoid `trapframe_set_elr` in other cases. The amount of ifdefs in trap.c is a bit concerning.